### PR TITLE
tables: Fix apt_sources and refactor tests

### PR DESCRIPTION
**Pull request by theopolis**
The `apt_sources` table was failing integration tests locally. Upon inspection it seems the table is not complete. The implementation is "based" on the APT library parsing and thus can have errors (which it did).

I added test cases for the types of sources I've observed. This does not mean it's complete.

### CODE REVIEW

**Comment by theopolis** [comment]
Fixes #5804.

File: osquery/tables/system/tests/posix/apt_sources_tests.cpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      // "dists" is not used as a section.
+      {"https://pkg.osquery.io apt/source/",
+       "https://pkg.osquery.io",
+       "pkg.osquery.io apt source",
+       "pkg.osquery.io_apt_source"},
+      // trailing slashes are stripped from URI.s
+      {"https://pkg.osquery.io/// apt/source/",
+       "https://pkg.osquery.io///",
+       "pkg.osquery.io apt source",
+       "pkg.osquery.io_apt_source"},
```
Comment (Smjert) [inline osquery/tables/system/tests/posix/apt_sources_tests.cpp]: I would add a case for `{"https://", "https://", "", ""}` if we think it's valid or maybe return an error in the apt table source code.
Though I would still test that case, together with another section in this test that tests failures for empty line, no comment, no protocol (possibly checking also that the error is what we expect).

**Comment by Smjert** [comment]
Thanks for your work!
Looks good in general with minor nits.

File: osquery/tables/system/tests/BUCK [hunk 2]
```diff
@@ -75,96 +28,24 @@ osquery_cxx_test(
...
 )
 
 osquery_cxx_test(
-    name = "shell_history_tests",
+    name = "posix_tests",
     platform_srcs = [
         (
             POSIX,
             [
+                "posix/apt_sources_tests.cpp",
```
[the remark this answers is not in the document]
Comment (theopolis) [reply osquery/tables/system/tests/BUCK]: I can make this change in the PR, but we do crazy things like install `yum` and `rpm` on macOS. I wanted to support as many use cases as possible since it was trivial (no dependencies needed in this case).

File: osquery/tables/system/tests/BUCK [hunk 3]
```diff
@@ -75,96 +28,24 @@ osquery_cxx_test(
...
-    ],
-)
-
-osquery_cxx_test(
-    name = "yum_sources_tests",
-    platform_srcs = [
-        (
-            POSIX,
-            [
                 "posix/yum_sources_tests.cpp",
```
[the remark this answers is not in the document]
Comment (theopolis) [reply osquery/tables/system/tests/BUCK]: Nope, please keep `yum` in `POSIX`!

File: osquery/tables/system/tests/posix/apt_sources_tests.cpp (hunk 1 above)
Comment (theopolis) [reply osquery/tables/system/tests/posix/apt_sources_tests.cpp]: Sure thing, will improve the test coverage here.

File: osquery/tables/system/tests/BUCK (hunk 2 above)
[the remark this answers is not in the document]
Comment (Smjert) [reply osquery/tables/system/tests/BUCK]: Oh ok then, I mean I knew that the (non-existing) package manager wasn't that good, but I didn't know you could resort to that! :D

**Comment by nabilschear** [comment]
this code is still pretty fragile and doesn't correctly handle sources.list files that conform to the spec listed here: http://manpages.ubuntu.com/manpages/xenial/man5/sources.list.5.html

Some examples of areas where it does not conform:
- more than one option will cause it to skip that line: `deb     [arch=amd64 trusted=yes] https://url bionic main`
- it only supports `http://` uris (no ftp, file, etc.) or more aptly it only supports a uri that has `://` in it.
- it names the repository to the second to last component name (e.g., name = `us.archive.ubuntu.com/ubuntu bionic universe`) in `deb [arch=amd64] http://us.archive.ubuntu.com/ubuntu/             bionic            main restricted universe multiverse`.  which doesn't make any sense.

there are some other things it doesn't support well.  So, I recommend that this code be updated based on the specification listed above and tested using the examples provided on the linked page.

**Comment by theopolis** [comment]
Hopefully moving the implementation closer to spec conformance is facilitated by the current unit test harnesses. For example, I hope we can add the examples from the spec to the current tests and then make small parsing changes such that they pass.

> it names the repository to the second to last component name

It is possible to deprecate the `name` (move it to a `HIDDEN` column by adding this attribute) and adopt the suite/components convention.

> A few fixes implemented [...]

Woot! I am happy to review the C++ code if you submit a PR (and I can do this quickly). I might not have time to try my hand at authoring the needed improvements for a while.

**Comment by nabilschear** [comment]
ok, ok, i'll do it proper.  Is changing the table format in scope here (moving to suite component) or should I just try to make it work within the current schema?

**Comment by theopolis** [comment]
Definitely in scope if it’s only 1 or 2 column name changes.
